code/modules/mob/mob_movespeed.dm

```diff
@@ -0,0 +1,158 @@
+/*! How move speed for mobs works
+
+Move speed is now calculated by using a list of movespeed modifiers, which is a list itself (to avoid datum overhead)
+
+This gives us the ability to have multiple sources of movespeed, reliabily keep them applied and remove them when they should be
+
+THey can have unique sources and a bunch of extra fancy flags that control behaviour
+
+Previously trying to update move speed was a shot in the dark that usually meant mobs got stuck going faster or slower
+
+This list takes the following format
+
+```Current movespeed modification list format:
+		list(
+			id = list(
+				priority,
+				flags,
+				legacy slowdown/speedup amount,
+				conflict
+			)
+		)
+```
+
+WHen update movespeed is called, the list of items is iterated, according to flags priority and a bunch of conditions
+this spits out a final calculated value which is used as a modifer to last_move + modifier for calculating when a mob
+can next move
+
+Key procs
+* [add_movespeed_modifier](mob.html#proc/add_movespeed_modifier)
+* [remove_movespeed_modifier](mob.html#proc/remove_movespeed_modifier)
+* [has_movespeed_modifier](mob.html#proc/has_movespeed_modifier)
+* [update_movespeed](mob.html#proc/update_movespeed)
+*/
+
+//ANY ADD/REMOVE DONE IN UPDATE_MOVESPEED MUST HAVE THE UPDATE ARGUMENT SET AS FALSE!
+
+///Add a move speed modifier to a mob
+/mob/proc/add_movespeed_modifier(id, update = TRUE, priority = 0, flags = NONE, override = FALSE, multiplicative_slowdown = 0, conflict = FALSE)
+	var/list/temp = list(priority, flags, multiplicative_slowdown, conflict) //build the modification list
+	var/resort = TRUE
+	if(LAZYACCESS(movespeed_modification, id))
+		var/list/existing_data = movespeed_modification[id]
+		if(!override)
+			return FALSE
+		if(priority == existing_data[MOVESPEED_DATA_INDEX_PRIORITY])
+			resort = FALSE // We don't need to re-sort if we're replacing something already there and it's the same priority
+		if(movespeed_modifier_identical_check(existing_data, temp))
+			return FALSE
+	LAZYSET(movespeed_modification, id, temp)
+	if(update)
+		update_movespeed(resort)
+	return TRUE
+
+///Remove a move speed modifier from a mob
+/mob/proc/remove_movespeed_modifier(id, update = TRUE)
+	if(!LAZYACCESS(movespeed_modification, id))
+		return FALSE
+	LAZYREMOVE(movespeed_modification, id)
+	UNSETEMPTY(movespeed_modification)
+	if(update)
+		update_movespeed(FALSE)
+	return TRUE
+
+///Handles the special case of editing the movement var
+/mob/vv_edit_var(var_name, var_value)

```

Comment:
I think that there's already a mob edit var, could you combine those for code quality?